### DISCUSSION TITLE
v2.3.0

### DISCUSSION THREAD

**Discussion by brianch** [brianch/offline-chess-puzzles#12] [Announcements]
As always, download the lichess puzzle database [from here](https://database.lichess.org/#puzzles) and unpack it to the "puzzles" directory of the app.

I'm rushing this release a bit because of the last point in the list below, the lichess db changed it's opening field and this required a change in this app as well. If you experience any issue, please let me know.

Thanks to lichess user Instit for contributing the French translation! Translations to other languages are welcomed.

## What's new:
- The interface is now translated to Portuguese, Spanish and French
- Added the option to show board coordinates
- Added the option to favorite/unfavorite puzzles and search these favorites
- Added a previous puzzle button
- Fixed compatibility with the current lichess db (they changed how the opening info is stored and so the previous offline-chess-puzzles executable wasn't working properly for this kind of search anymore)

<hr /><em>This discussion was created from the release <a href='https://github.com/brianch/offline-chess-puzzles/releases/tag/v2.3.0'>v2.3.0</a>.</em>

**Comment by travelgirl-fics** [comment]
question regarding engine -- have produced a link to stockfish 16 (within lucaschess), but i don't see evidence your program sees the engine (no errors, but no actions, either).  guessing i have missed initialising and/or have a parameter or three wrong.

love the program as-is.  works really nicely;  simple, clean, fast.  thank you.

**Reply by brianch** [reply]
Hello!

> question regarding engine -- have produced a link to stockfish 16 (within lucaschess), but i don't see evidence your program sees the engine (no errors, but no actions, either). guessing i have missed initialising and/or have a parameter or three wrong.

Did you remember to include the extension in the "engine path" field? In that field you should have something like this: `C:\stockfish16\stockfish16_win_x64_avx2\stockfish16_x64_avx2.exe`

In the future I definitely should include a dialog window to select the file and make this easier 😅

> love the program as-is. works really nicely; simple, clean, fast. thank you.

Thank you so much! This is exactly the goal of the project :)

**Comment by travelgirl-fics** [comment]
interesting.  i tried several different versions (all from the lucaschess
local folders).  as long as the URI was surrounded by quotation marks, the
engine didn't appear to start.  take away the quotes, and it ran,
displaying only the one move it deemed "best" at that moment.  single
quotes also fail to work.

not sure why having quotations (which would normally need to be in place
when URI strings contained spaces, for example) would make the engine (or
the display of the best move) fail...  checking to see if the engine, once
instantiated, ever removes itself from memory when the button STOP ENGINE
is clicked...  it does, according to process explorer...

having said that, i was looking for something a bit more obvious than a
small line underneath MODE ANALYSIS.  perhaps a separate pop-up window
containing a few lines, not just the "best".  or, a larger font in a
dedicated location on-screen (current puzzle tab?)

anyway, problem solved.  thanks  :)

**Reply by brianch** [reply]
Ah, nice to see you sorted it out!

Integrating a file chooser dialog to select the engine file would avoid these unclear things, I'll try to do it in a future version.

> having said that, i was looking for something a bit more obvious than a
small line underneath MODE ANALYSIS.  perhaps a separate pop-up window
containing a few lines, not just the "best".  or, a larger font in a
dedicated location on-screen (current puzzle tab?)

You can make moves on the board (and take them back) to see how the eval changes, the logic for this is that I kinda feel it's a better way to learn than to have the moves right away in front of us, I also didn't want to make the engine standout too much, it was meant more as a quick way to check another line (and of course, I ain't gonna lie, it's like that also because it would be much more work to do otherwise 😆).

If we were to include more info, I guess a new tab (that only appears when the engine is on) would be better, but I do believe having to input the moves is a nice approach for a learning tool.

**Comment by travelgirl-fics** [comment]
i'd like to suggest an enhancement:  the ability to save off individual
puzzles to a separate database, as you are playing.  this would be
important so you could re-test those you got wrong (or had problems with,
but got correct), so you can go over them at a later date/time...  i'd
prefer pgn, but the data from the CSV would be fine as well...

thanks :)

**Reply by brianch** [reply]
Yeah, that's quite important indeed. For now you can add them to the favorites and search the favorites afterwards, instead of the lichess DB.

The plan was that this would be a first step in perhaps being able to have different user-created collections of puzzles (like "beautiful" ones, "things I got wrong", "counter-intuitive puzzles", etc.) instead of just a general 'favorites". But this needs some planing, CSV would be quite slow to write to if there were too many puzzles, and an actual database, like I'm currently using for favs, is a bit risky since lichess can change the CSV format at any time (they did it this year for example) which would break things.

Did you mean a PNG image? PGN is for the notation of whole games. I actually got an 'export search to PDF' function working, which should be available in the next release, but I'm a bit busy this month with other stuff, so it may take a while still.

**Comment by travelgirl-fics** [comment]
pretty sure i said PGN, not PNG, but that's ok, we both knew what we said :)

i've worked with computers since the late 60s, and started owning them in
the mid 70s...  if you need someone for testing, or to bounce ideas off,
let me know...  i've been retired for a few years, and occasionally have
time on my hands for other projects :)

oh.  where are the FAVORITES stored, and what format?  how to
access/maintain/delete/save/etc ?  i know you can choose the favorites to
run through, but right now, (1) it's only one puzzle, and (2) it was just a
test, not serious about saving it....  :)

**Reply by brianch** [reply]
The favorites are being stored in a sqlite database (it most probably will stay that way, but there's a chance I might change this in the future).

If the current puzzle is already in the favorites you will see a button to delete it from there, and if it's not, a button to add it, other than that there's no way to go thought them currently (well, except opening the sqlite db itself and changing it), but at least there's already the guarantee that it won't have duplicates.

And thanks a lot for the offer to help! I'll keep it in mind, always great to have people who can test :) And I hope you're enjoying life after the retirement :)

By the way, do you happen to have a Mac? Checking if it runs there is one of the things I cannot do myself.

**Comment by travelgirl-fics** [comment]
nope...  i was an early apple ][ user, but had problems with how expensive
apples were LATER -- the lisa, every mac since, all have been 2-3 times
what a comparable PC was...

if you want it to stay in sqlite, and that's not a bad idea, i'd write some
code to allow maintenance (though i'm hard-pressed to know what maintenance
besides DEL would be useful...  maybe copying/pasting test positions from
one database to another?

**Reply by brianch** [reply]
> nope...  i was an early apple ][ user, but had problems with how expensive
apples were LATER -- the lisa, every mac since, all have been 2-3 times
what a comparable PC was...

Yeah, it's really unreasonably expensive. As I'm a bit younger, I never saw this not being the case, so I never had any interest in having one myself.

> if you want it to stay in sqlite, and that's not a bad idea, i'd write some
code to allow maintenance (though i'm hard-pressed to know what maintenance
besides DEL would be useful...  maybe copying/pasting test positions from
one database to another?

And oh, I don't think there's much need right now to work more on this, I think it's already very usable.
